concent_api/core/tests/test_payments_bankster.py

```diff
@@ -417,14 +417,14 @@ def test_that_settle_overdue_acceptances_should_return_claim_deposit_with_amount
                         timestamp="2018-02-04 9:00:05",
                         task_to_compute=task_to_compute
                     ),
-                    payment_ts="2018-02-04 10:00:05",
+                    payment_ts="2018-02-04 9:00:05",
                 ),
                 self._get_deserialized_subtask_results_accepted(
                     report_computed_task=self._get_deserialized_report_computed_task(
                         timestamp="2018-02-04 23:00:05",
                         task_to_compute=task_to_compute
                     ),
-                    payment_ts="2018-02-05 10:00:05",
+                    payment_ts="2018-02-04 10:00:05",
```

Comment:
Is this change necessary and why? BTW, timestamp of created above `report_computed_task` is 2018-02-04 23:00:05, so I think it is wrong to set `payment_ts` as You did.. It should be later

Reply:
Newest `golem_messages` does not allow to create `SubtaskResultsAccepted` with `payment_ts` from future. That's why here and in all others `payment_ts` changes it is changed that way.

Reply:
OK.. But still we have freeze_time("**2018-02-04 10:00:15**") and timestamp="**2018-02-04 23:00:05**" in one of `report_computed_task`. This should fail for me, how can someone send at **10:00:15** `report_computed_task` with timestamp "**23:00:05**" (both in the same day). 
@kbeker You didn't change this times, so thanks for explanation. :+1: 
Should we validate such situations or leave it as it is now? @cameel ?

Reply:
`RCT.timestamp > SRA.payment_ts` would be weird but it should be very rare because it would require one of them to be in the future or in the past when sent. Clients validate that received messages are not too new or too old so only a modified client could do that. And with a modified client our only concern is whether it can break something when it's malicious.

I don't think it breaks anything. We have safeguards against `payments_ts` too far in the past where it matters (e.g. #1036). RCT cannot be too far in the future either because it must be earlier than the deadline.

Such weird situations with timestamps are very often possible with nested messages. My approach here is not to try to validate all the relationships (that would be best but it's too complex because we can never treat timestamps as exact and there are so many of them). Just validate the things that can cause problems.

Reply:
I agree with @pawelkisielewicz - such a timestamp of the 2nd `report_computed_task` doesn't make sense... please modify it to be more reasonable